src/main/java/com/ericsson/eiffel/semantics/events/Event.java

```diff
@@ -55,24 +55,7 @@ public Meta generateMeta(Meta meta) {
      */
     public String getFamilyRoutingKey(String eventType) {
         try {
-            return RoutingKeyTypes.valueOf(eventType).getFamily();
-        } catch (Exception e) {
-            log.error("Unknown event type requested: " + eventType);
-        }
-        return null;
-    }
-
-    /**
-     * This method is used to get type based on the eventType
-     * 
-     * @param eventType
-     *            Eiffel eventType
-     * @return type based on eventType (or) error will be thrown if unknown
-     *         eventType requested.
-     */
-    public String getTypeRoutingKey(String eventType) {
-        try {
-            return RoutingKeyTypes.valueOf(eventType).getType();
+            return RoutingKeyFamilies.valueOf(eventType).getFamily();
         } catch (Exception e) {
```

Comment:
Do we really need to catch a general exception? What if NullPointerException, ArrayIndexOutOfBoundsException is thrown in valueOf() or getFamily()? Then we incorrectly report "Unknown event type...".

Reply:
Resolved

Reply:
We really should not catch general exception here. Catch only exceptions declared by method valueOf(), i.e. IllegalArgumentException and NullPointerException.